CI failed frequently because of building image failure

**Describe the bug**
Similar to bug #48, CI tests failed because of the following image failure:
```
Step 15/17 : COPY --from=cni-binaries /opt/cni/bin /opt/cni/bin
invalid from flag value cni-binaries: No such image: sha256:9b6d192c5e82ad41f727a3c5b6ec95d699a4a27fb1b38e24c95e89cd3e7138d3
Makefile:178: recipe for target 'build-ubuntu' failed
make: *** [build-ubuntu] Error 1
```
The missing image was made in step 4:
```
Step 4/17 : RUN mkdir -p /opt/cni/bin &&     wget -q -O - https://dl.k8s.io/network-plugins/cni-plugins-amd64-v0.7.5.tgz | tar xz -C /opt/cni/bin $CNI_PLUGINS
 ---> Running in 94adf35da9ac
Removing intermediate container 94adf35da9ac
 ---> 9b6d192c5e82
```
but was cleaned up by other builds executed simultaneously:
https://github.com/vmware-tanzu/antrea/blob/ab3414e4d557315bd60d133b17f53dde4834b5f1/ci/jenkins/jobs/macros.yaml#L182.

**To Reproduce**
Seen in jenkins logs:
https://jenkins.antrea-ci.rocks/job/antrea-e2e-for-pull-request/118/console
https://jenkins.antrea-ci.rocks/job/antrea-e2e-for-pull-request/117/console
https://jenkins.antrea-ci.rocks/job/antrea-e2e-for-pull-request/99/console
https://jenkins.antrea-ci.rocks/job/antrea-networkpolicy-for-pull-request/74/console
https://jenkins.antrea-ci.rocks/job/antrea-conformance-for-pull-request/52/console
https://jenkins.antrea-ci.rocks/job/antrea-conformance-for-pull-request/50/console

**Expected**
CI should build images successfully consistently.

**Actual behavior**
CI failed quite often because of this failure.

--- a/ci/jenkins/jobs/macros.yaml
+++ b/ci/jenkins/jobs/macros.yaml
@@ -179,7 +179,9 @@
           make clean
           docker pull antrea/openvswitch --all-tags
           docker images | grep 'antrea-ubuntu' | awk '{print $3}' | xargs -r docker rmi || true
-          docker images | grep '<none>' | awk '{print $3}' | xargs -r docker rmi || true
+          # Clean up dangling images generated in previous builds. Recent ones must be excluded
+          # because they might be being used in other builds running simultaneously.
+          docker image prune -f --filter "until=1h"
           VERSION="$JOB_NAME-$BUILD_NUMBER" make
 
           sed -i "s|#serviceCIDR: 10.96.0.0/12|serviceCIDR: 100.64.0.0/13|g" build/yamls/antrea.yml

